From: Jie Zhang <jie.zhang@analog.com>
Date: Mon, 1 Jun 2009 07:25:42 +0000 (+0000)
Subject: Backport UCLIBC_HAS_STUBS config option from the upstream. This option makes uClibc... 
X-Git-Url: http://blackfin.uclinux.org/git/?p=readonly-mirrors%2Ftoolchain.git;a=commitdiff_plain;h=cacea6b12bae6f6cfd08e2bbec536e9b23b342f5

Backport UCLIBC_HAS_STUBS config option from the upstream. This option makes uClibc provide fork() stub on NOMMU targets. It's default off.


git-svn-id: svn://localhost/svn/toolchain/trunk@3417 2adf4935-75e9-4407-93b2-34b21a1880a4
---

Index: uClibc-0.9.30.2/libc/unistd/daemon.c
===================================================================
--- uClibc-0.9.30.2.orig/libc/unistd/daemon.c	2010-01-13 22:22:31.000000000 +0100
+++ uClibc-0.9.30.2/libc/unistd/daemon.c	2010-04-06 22:04:11.000000000 +0200
@@ -55,7 +55,6 @@
 libc_hidden_proto(dup2)
 libc_hidden_proto(setsid)
 libc_hidden_proto(chdir)
-libc_hidden_proto(fork)
 
 #ifndef __ARCH_USE_MMU__
 #include <sys/syscall.h>
@@ -85,6 +84,8 @@
 	return ret;
 }
 #else
+libc_hidden_proto(fork)
+
 static inline pid_t fork_parent(void)
 {
 	switch (fork()) {
